### PR TITLE
common orm and reactive persistence tests #367

### PR DESCRIPTION
**Pull request by blafond**

### CODE REVIEW

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+		Session ormSession = ormFactory.openSession();
+		ormSession.beginTransaction();
+		ormSession.persist( rose );
+		ormSession.getTransaction().commit();
+		ormSession.close();
+
+		// Check database with Mutiny session and verify 'rose' flour exists
+		Mutiny.Session mutinySession = openMutinySession();
+		// find the entity using the id
+		test( context, mutinySession.find( Flour.class, rose.id)
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java:100]: > // find the entity using the id
> test( context, mutinySession.find( Flour.class, rose.id)

I'm not a big fun of comments describing exactly the signature of the method. I think you can remove it.

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+				// and check it's the right one
+				.invoke( foundRose -> context.assertEquals( rose, foundRose ) )
+				// and that it's not the wrong one
+				.invoke( noAlmond -> context.assertNotEquals(rose, sunflower))
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java:104]: Is it really useful to test that the two objects are different? You have already checked that rose is the expected one.
The only way this assert can fail is if there is something wrong in the test.
It doesn't seem very helpful.

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+			  	// Close session
+				.eventually(mutinySession::close)
+				// verify session closed
+				.invoke( sessionClosed -> context.assertFalse(mutinySession.isOpen()) )
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java:108]: No need to close the session and it. The super class takes care of it.

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import io.vertx.ext.unit.TestContext;
+
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.POSTGRESQL;
+import static org.hibernate.reactive.util.impl.CompletionStages.completedFuture;
+
+public class ORMReactivePersistenceTest extends BaseReactiveTest {
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java:33]: I think in this case a brief javadoc explaining what this test is about is useful

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+		StandardServiceRegistryBuilder builder = new StandardServiceRegistryBuilder()
+				.applySettings( configuration.getProperties() );
+
+		StandardServiceRegistry registry = builder.build();
+		ormFactory = configuration.buildSessionFactory( registry );
+	}
+
+	@After
+	public void cleanDb(TestContext context) {
+		ormFactory.close();
```
Comment (DavideD) [inline hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java:62]: If an error happens here the db doesn't get cleaned. Maybe we should add a try catch. I think we can ignore the error print it (if an error occurs here is probably due to something else).

**Review by DavideD — CHANGES_REQUESTED** [review]

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java (hunk 4 above)
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java:33]: Something like this?

/**
 * These tests verify that data can be persisted and queried on the same database
 * using both JPA/hibernate and reactive Mutiny or CompletionStage session factories
 */

Comment (DavideD) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java:33]: Maybe something like this:
```
Test that it's possible to use an ORM and Reactive session in the same app
<p>
The test creates an entity using the ORM session and then retrieve it using the reactive one.
```

I don't have strong opinion, feel free to pick the one you prefer.

File: hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java (hunk 5 above)
Comment (blafond) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java:62]: print as in....      **context.fail(exception);**     ?

Comment (DavideD) [reply hibernate-reactive-core/src/test/java/org/hibernate/reactive/ORMReactivePersistenceTest.java:62]: Actually, forget about it. We clean the data when we re-run the tests so it's fine the way it is

**Comment by DavideD** [comment]
Rebased and Merged https://github.com/hibernate/hibernate-reactive/pull/402

**Pull request closed without merging**